packagingbuild/xenial/Dockerfile

```diff
@@ -38,22 +38,24 @@ RUN echo -e "#!/bin/sh\nexit 101\n" > /usr/sbin/policy-rc.d && \
 #
 
 
-# install python development
-RUN apt-get update && \
-    apt-get -y install build-essential python-dev python python-virtualenv
+# Install python 3.6 & development
+# from the PPA as it's not available in base distro
+RUN sudo add-apt-repository ppa:deadsnakes/ppa && \
+    apt-get update && \
+    apt-get -y install build-essential python3.6-dev python3.6
 
 
 RUN apt-get update && \
     apt-get -y install \
         devscripts debhelper dh-make libldap2-dev libsasl2-dev && apt-get clean
 
 # Install fresh pip and co
-RUN curl https://bootstrap.pypa.io/get-pip.py | python - virtualenv==16.6.0 pip==19.1.1 wheel setuptools; \
-      pip install --upgrade requests[security] && rm -rf /root/.cache
+RUN curl https://bootstrap.pypa.io/get-pip.py | python3.6 - virtualenv==16.6.0 pip==19.1.1 wheel setuptools; \
```

Comment:
Normally I wouldn't want to ping to a specific point release of Python, but I understand why you have to given the Ubuntu 16.04 gotchas you listed (the main one being: `python3` is Python 3.5).

I don't expect this to be an issue for future ST2 releases since we will be dropping Ubuntu 16.04 support after ST2 3.4 anyway.

Reply:
^ There's nothing to change here, I just wanted to explain why this was okay, to my future self or to others. Resolving.